### ISSUE TITLE
sbt launcher should look at the `no_proxy` enviornment variable

### ISSUE THREAD

**Issue by jsuereth** [sbt/sbt#1605]
It's a default in linux and should be translated into the `htttp.nonProxyHosts` property

**Comment by saurfang** [comment]
+1. I recently struggled about this on Windows because Windows didn't like `|` in `-Dhttp.nonProxyHosts=localhost|127.0.0.1|*.foo.com` for `JAVA_OPTS` or `SBT_OPTS`. It turns out I can use `^` to escape `|` but then if `NO_PROXY`, which is comma separated, can be picked up directly, it would be very helpful.

**Comment by fatlazycat** [comment]
Is there anyway at all to get SBT to not use the proxy for certain hosts ?

**Comment by zoosky** [comment]
Same question as @fatlazycat 
sbt 0.13.13 seems to ignore nonProxyHosts settings in Windows and Linux and the `no_proxy` enviornment variable. 

How do you work with a local nexus repository in  an enterprise setting? Any suggestions?

**Comment by zoosky** [comment]
@eed3si9n What is needed to fix this?

**Comment by eed3si9n** [comment]
@zoosky Any research into what component is broken would be helpful.
Is the launcher that launches sbt broken or the Ivy part that's used for `update`?

**Comment by zoosky** [comment]
@eed3si9n I see the issue when sbt downloads libraries from repositories. It can't find the intranet nexus. Is this in Ivy?

**Comment by eed3si9n** [comment]
Yes. That portion is Ivy, and it's relatively easier to tackle than updating the launcher.

**Comment by eed3si9n** [comment]
For example, if you use alternative dependency resolver Coursier (uses OkHttp internally iirc) would you still see the issue?

**Comment by zoosky** [comment]
@eed3si9n I'll try Coursier, seems easier than fixing upstream Ivy.

**Comment by zoosky** [comment]
@eed3si9n just tried with sbt-coursier 1.0.0-RC1. Nice, but does not solve the issue. It can't find the POM URL because it goes out via proxy.
